src/corelibs/U2Core/src/tasks/CmdlineTaskRunner.cpp

```diff
@@ -104,6 +104,7 @@ bool containsPrefix(const QStringList &args, const QString &prefix) {
 }    // namespace
 
 const QString CmdlineTaskRunner::REPORT_FILE_ARG = "ugene-write-task-report-to-file";
+const QString CmdlineTaskRunner::ERROR_STATE_FILE_ARG = "ugene-write-task-error-state-to-file";
```

Comment:
Why stderr is not enough in our case?
Using stderr + exit code is a standard way how command line tools report that something went wrong

Reply:
The process really could return you an exit code, but just the number - no details. Of course, you can have a list of errors for each return code, but I don't see any advantages of this method. I use the infrastructure which has already been used for Sanger reports - use the temporary file to save the output of the process, this solution has been introduced by @atiunov, and I like it more that your idea.

Anyway, I won't change the good solution of this issue (or any other issue) just because "you don't like it" or "you have your own idea how it should be done" - take it in account in future.

Reply:
> Anyway, I won't change the good solution of this issue (or any other issue) just because "you don't like it" or "you have your own idea how it should be done" - take it in account in future.

It was only a question why the standard command-line solution is not used here and you selected a more complex and advanced non-standard variant.
Since I do not see the answer for the problem and only 'take it in account' statement above I'm closing this PR and re-assigning the bug to myself.
Thank you.

Reply:
I used this solution because it's been already used for [the similar issue](https://ugene.dev/tracker/browse/UGENE-5591). Your solution is new and the new solution is more difficult to implement that the solution which has been already use (you have said that "I selected a more complex and advanced non-standard variant", but it's not true).

It was no certain solution had to be implemented (nothing about it in the issue) and, believe it or not, but I can't read your minds. I spent a day to this bug fix - it works and it works good. So, I think, we are done here and this conversation could be resolved.

Reply:
OK, I can accept your solution here if you insist.

Just a couple of questions to mark it resolved:

1) Do you think that "I spent a day to this bug fix - it works and it works good - and I'm not about to re-make it." is a good technical reason may be used to compare 2 different options? 

2)  You say "I can't read your minds", but I can't neither: I can't give my opinion about the solution before you present it. And once you present it you consider it as 'no discussion is possible'. How this situation can be avoided in the future?

3) If you have a command line utility, which behavior do you expect from it on error:
 A) Return non-0 (error) exit code and error message into the error output stream.
 B) Return 0 (success) exit code and only report an error if there is a special 'error-output-file' parameter.

That is all I have to say about current discussion. I will mark this thread resolved and approve your variant. Please address all other minor issues.